rcsb/workflow/wuw/PdbCsmImageWorkflow.py

```diff
@@ -0,0 +1,154 @@
+            bcif_file = args["csm_base_dir"] + model_path
+            if Path.exists(bcif_file):
+                t1 = Path.stat(bcif_file).st_mtime
+                t2 = metadata["datetime"].timestamp()
+                if t1 < t2:
+                    model_list.append(f"{model_id} {model_path} computational")
+            else:
+                model_list.append(f"{model_id} {model_path} computational")
+    return model_list
+
+
+def images_gen_lists(args: dict) -> None:
+    """Generate lists of pdbs/csms in files."""
+    pdb_id_list = get_pdb_list(args)
+    comp_id_list = [] if args["imgs_exclude_models"] else get_csm_list(args)
+
+    # Print results, combine, and shuffle
+    if len(pdb_id_list) < 1 and len(comp_id_list) < 1:
+        msg = "pdb and csm id list empty"
+        raise ValueError(msg)
+    msg = f"There are {len(pdb_id_list)} pdb_ids and {len(comp_id_list)} comp_ids for which to generate BCIF and images"
+
+    full_id_list = pdb_id_list + comp_id_list
+    random.shuffle(full_id_list)
+
+    steps = int(len(full_id_list) / int(args["num_workers"]) )
+    for i in range(0, len(full_id_list), steps):
+        Path(args["id_list_path"]).mkdir(parents=True, exist_ok=True)
+        with Path.open(args["id_list_path"] + str(int(i/steps)), "w") as f:
+            for line in full_id_list[i : i + steps]:
+                f.write(line + "\n")
+
+def images_gen_jpgs(args: dict) -> None:
+    """Generate jpgs for given pdb/csm list."""
+    with Path.open(args["id_list_path"] + args['file_number'], 'r') as f:
+        id_list = [line.rstrip('\n') for line in f]
+    if not isinstance(id_list, list):
+        msg = "id_list not a list"
+        raise TypeError(msg)
+    
+    for line in id_list:
+        # Requirements:
+        # 1. bcif files must be unzipped
+        # 2. bcif files must be in a local dir (otherwise I'll have to add a curl step)
+        # 3. bcif files do not need conversion from cif files. This should be taken care of in bcif workflow
+        # 
+        # todo: add a step that checks:
+        # 1. the dir exists
+        # 2. the file is a bcif with data inside of it
+        # 3. the output dir is availible
+
+        file_id, bcif_file_name, sdm = line.split(" ")
+        content_type_dir = "pdb/" if sdm == "experimental" else "csm/"
+
+        bcif_file_path = args["pdb_base_dir"] + bcif_file_name if sdm == "experimental" else args["csm_base_dir"] + bcif_file_name 
+        
+        out_path = args["jpgs_out_dir"] + content_type_dir
+        Path(out_path).mkdir(parents=True, exist_ok=True)
+        
+        ### run_molrender ###
+        cmd = [
+            '/usr/bin/xvfb-run',
+            '-a',
+            '-s', '-ac -screen 0 1280x1024x24',
+            args["molrender_exe"],
+            'all',
+            bcif_file_path,
+            out_path,
+            '--height', '500',
+            '--width', '500',
+            '--format', 'jpeg'
```

Comment:
Since this is now a more general CLI outside the weekly update codebase I think it'd be important to extract the different parameters as CLI parameters, e.g. image size, xvfb executable path (possibly an optional parameter, because some environments may not require xvfb), etc.

Reply:
I just added them!